### PR TITLE
feat: set up org

### PR DESCRIPTION
**Pull request by livio-a**

### CODE REVIEW

File: internal/v2/domain/iam.go [hunk 1]
```diff
@@ -11,7 +11,7 @@ type IAM struct {
 	IAMProjectID                    string
 	SetUpDone                       Step
 	SetUpStarted                    Step
-	Members                         []*IAMMember
+	Members                         []*Member
 	IDPs                            []*IDPConfig
 	DefaultLoginPolicy              *LoginPolicy
 	DefaultLabelPolicy              *LabelPolicy
```
Comment (fgerschwiler) [inline internal/v2/domain/iam.go:17]: Do we need this on iam struct?

File: internal/v2/domain/org.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+	Domains                  []*OrgDomain
+	Members                  []*Member
+	OrgIamPolicy             *OrgIAMPolicy
+	LoginPolicy              *LoginPolicy
+	LabelPolicy              *LabelPolicy
+	PasswordComplexityPolicy *PasswordComplexityPolicy
+	PasswordAgePolicy        *PasswordAgePolicy
+	PasswordLockoutPolicy    *PasswordLockoutPolicy
+	IDPs                     []*IDPConfig
```
Comment (fgerschwiler) [inline internal/v2/domain/org.go:23]: Do we need this on Org?

File: internal/v2/repository/org/domain.go [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+	}
+}
+
+func DomainAddedEventMapper(event *repository.Event) (eventstore.EventReader, error) {
+	orgDomainAdded := &DomainAddedEvent{
+		BaseEvent: *eventstore.BaseEventFromRepo(event),
+	}
+	err := json.Unmarshal(event.Data, orgDomainAdded)
+	if err != nil {
+		return nil, errors.ThrowInternal(err, "USER-GBr52", "unable to unmarshal org domain added")
```
Comment (fgerschwiler) [inline internal/v2/repository/org/domain.go:50]: wrong err ids in whole file (USER --> ORG)

File: internal/v2/repository/org/org.go [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+	}
+}
+
+func OrgAddedEventMapper(event *repository.Event) (eventstore.EventReader, error) {
+	orgAdded := &OrgAddedEvent{
+		BaseEvent: *eventstore.BaseEventFromRepo(event),
+	}
+	err := json.Unmarshal(event.Data, orgAdded)
+	if err != nil {
+		return nil, errors.ThrowInternal(err, "USER-Bren2", "unable to unmarshal org added")
```
Comment (fgerschwiler) [inline internal/v2/repository/org/org.go:46]: wrong err ids in whole file (USER --> ORG)

File: internal/admin/repository/org.go [hunk 5]
```diff
@@ -2,21 +2,17 @@ package repository
...
 )
 
 type OrgRepository interface {
-	SetUpOrg(context.Context, *admin_model.SetupOrg) (*admin_model.SetupOrg, error)
 	IsOrgUnique(ctx context.Context, name, domain string) (bool, error)
 	OrgByID(ctx context.Context, id string) (*org_model.Org, error)
 	SearchOrgs(ctx context.Context, query *org_model.OrgSearchRequest) (*org_model.OrgSearchResult, error)
 
 	GetOrgIAMPolicyByID(ctx context.Context, id string) (*iam_model.OrgIAMPolicyView, error)
-	GetDefaultOrgIAMPolicy(ctx context.Context) (*iam_model.OrgIAMPolicyView, error)
```
[the remark this answers is not in the document]
Comment (livio-a) [reply internal/admin/repository/org.go:18]: this was redundant... query side uses IAMRepository

**Review by fgerschwiler — APPROVED** [review]

**Pull request merged**